src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/AuthorizationRepository.kt

```diff
@@ -0,0 +1,171 @@
+package org.vaccineimpact.orderlyweb.db
+
+import org.jooq.Record
+import org.pac4j.core.authorization.generator.AuthorizationGenerator
+import org.pac4j.core.context.WebContext
+import org.pac4j.core.profile.CommonProfile
+import org.vaccineimpact.orderlyweb.db.Tables.*
+import org.vaccineimpact.orderlyweb.errors.UnknownObjectError
+import org.vaccineimpact.orderlyweb.models.Scope
+import org.vaccineimpact.orderlyweb.models.permissions.PermissionSet
+import org.vaccineimpact.orderlyweb.models.permissions.ReifiedPermission
+import org.vaccineimpact.orderlyweb.security.authorization.orderlyWebPermissions
+
+interface AuthorizationRepository
+{
+    fun ensureUserGroupHasPermission(userGroup: String, permission: ReifiedPermission)
+}
+
+class OrderlyAuthorizationRepository : AuthorizationGenerator<CommonProfile>, AuthorizationRepository
+{
+    private val ALL_GROUP_PERMISSIONS = "all_group_permissions"
+    private val PERMISSION_NAME = "permission_name"
+    private val GROUP_PERMISSION_ID = "permission_id"
+
+    override fun generate(context: WebContext?, profile: CommonProfile): CommonProfile
```

Comment:
Would it be slightly cleaner to keep the repository separate from the pac4j AuthorizationGenerator class? Might mean a largely-boilerplate generator class though, so your call! If we do keep it as one class, I think I'd be inclined to rename it since it isn't a 'normal' repository (ah, although it does support a repo interface as well doesn't it). 

(I started writing a comment complaining about 'generate' being a confusing name before I realised it was an override because this was actually an AuthorizationGenerator class!)

Reply:
I've got one class implementing 2 interfaces and sharing a fair bit of code, so I have to say I do think it makes sense as is! Happy to change the name if you have a suggestion for a clearer one?

Reply:
in fact totally changed my mind, I'll separate them into 2 classes for clarity